### DISCUSSION TITLE
Good practices and tips&tricks to selecting/creating an initial guess

### DISCUSSION THREAD

**Discussion by VDB-Bram** [KULeuvenNeuromechanics/PredSim#170] [General]
Some first thoughts.

# Good practices
- Use multiple initial guesses. Chances of finding a local rather then the absolute minimum are high. Select the simulation with the lowest cost as 'optimal'.
- If your initial guess has an assymmetrical gait pattern, make sure it is a multiple of 1 gait cycle (i.e. 1, 2, 3 ... gait cycles). Do not forget to specify this in the settings (see [S.solver.IG_selection_gaitCyclePercent](https://github.com/KULeuvenNeuromechanics/PredSim/blob/master/Documentation/SettingsOverview.md#required-settings)).

# Tips & Tricks
* For better convergence, change e.g. pelvis_ty such that when evaluating the external function with the IG the GRF make sense. When the model is floating above the ground or deep in the ground at the initial guess, your simulations probably will need more itterations to converge.
* You can use one of your simulation results as a new initial guess. This is especially usefull when running simulations with personalizations where you can use the result of 'generic' model.

**Comment by tombuurke** [comment]
I noticed that setting S.subject.adapt_IG_pelvis_y = 1 results in poor convergence or weird solutions (e.g. skipping/jumping motions) in the data informed initial guess.
When I rerun some of the simulations with S.subject.adapt_IG_pelvis_y = 0, (most of) my issues are solved.

**Reply by VDB-Bram** [reply]
Do your models have leg length discrepancies? I have noticed that when there are such discrepancies this can lead to widely different GRFs at the evaluation of the initial guess (iteration 0) resulting in what you observed (poor convergence or weird solutions).

**Reply by Minh309** [reply]
I also had the same problems and I found that in the file getGuess_DI_opti.m, there are different way to formulate initial guess depending on S.subject.adapt_IG_pelvis_y variable. If you set S.subject.adapt_IG_pelvis_y = 0, the pelvis_ty value will be the same as the value from your intial guess. However, when you set S.subject.adapt_IG_pelvis_y = 1, the pelvis_ty value is now calculated differently as pelvis_ty = model_info.IG_pelvis_y + deviation, where the deviation value is calculated as guess.Qs(:,model_info.ExtFunIO.jointi.base_vertical) - mean(guess.Qs(:,model_info.ExtFunIO.jointi.base_vertical)).

My guess is that your model_info.IG_pelvis_y was too small, which means the model sinks too much into the ground, leading high GRF values. This will in turn generate high value of passive moments as well and likely cause your optimization to converge in local minima. I think increasing model_info.IG_pelvis_y will likely solve your issue.
